src/elementPlugins/ApplicationViewWrapper.tsx

```diff
@@ -1,15 +1,31 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { ErrorBoundary, pluginProvider } from './'
 import { ApplicatioViewProps, PluginComponents } from './types'
+import { Message } from 'semantic-ui-react'
+import evaluateExpression from '@openmsupply/expression-evaluator'
 
 const ApplicationViewWrapper = (props: ApplicatioViewProps) => {
   const {
     templateElement: { elementTypePluginCode: pluginCode },
-    isVisible,
+    isVisibleExpression,
   } = props
 
+const [isVisible, setIsVisible]:any[] = useState({visible: evaluateExpression(isVisibleExpression), error: null})
+
+useEffect(() => {
```

Comment:
@CarlosNZ, when would this be called ? i.e. would there be a prop to 'applicationState' on this component, and whenever anything changes in 'applicationState', React would call ApplicationViewWrapper which would in turn call useEffect ? What if nothing in applicationState changes this ApplicationViewWrapper (and it doesn't need to re render, would useEffect be called) ? Would be great to have an example of relying on some application state.

I think the reason i had it in parent component, was to use onUpdate method from any of the wrappers, to re run the evaluationExpression for visibility of all components. If it can be done within the wrapper, it's probably better

Reply:
Yeah, in this case it's only going to update when the actual _expression_ changes, which happens in the template builder, but not in an actual application.

In the real one, we'd also pass in the `{question_code: response}` object containing all the questions (plus the `user` object and any other  useful bits of local state), and monitor those objects with `useEffect`. That should work, shouldn't it?

Reply:
I wasn't sure exactly when useEffect would be called, I think it should be called any time it re-renders though 👍